google-http-client/src/main/java/com/google/api/client/util/escape/CharEscapers.java

```diff
@@ -90,6 +92,28 @@ public static String decodeUri(String uri) {
     }
   }
 
+  /**
+   * Decodes the path component of a URI. This must be done via a method that does not try to
+   * convert + into spaces(the behavior of {@link java.net.URLDecoder#decode(String, String)}). This
+   * method will transform URI encoded value into their decoded symbols.
+   *
+   * <p>i.e: {@code decodePath("%3Co%3E")} would return {@code "<o>"}
+   *
+   * @param path the value to be decoded
+   * @return decoded version of {@code path}
+   */
+  public static String decodeUriPath(String path) {
+    if (path == null) {
+      return null;
+    }
+    try {
+      return URLDecoder.decode(path.replace("+", "%2B"), "UTF-8");
```

Comment:
I think you can use Charsets.UTF_8 or some such and avoid the try catch

Reply:
Charset api requires 10+.

Reply:
StandardCharsets.UTF_8 is in java 7.

Reply:
I meant the api that accepts a Charset, but I suppose I can call `name()` on it